_kbarticles/2022-11-04-fixing-certificate-errors-aws-egress-proxy.md

```diff
@@ -0,0 +1,18 @@
+---
+layout: post
+title: "Fixing certificate validation errors from AWS CLI when using the cg-egress-proxy"
+date: November 4, 2022
+excerpt: How to fix certificate errors if you are using the AWS CLI from an application protected by the cg-egress-proxy
+---
+
+If you are using the [`cg-egress-proxy`](https://github.com/GSA/cg-egress-proxy) to restrict the egress traffic for your application, you may experience certificate validation errors which prevent you from [interacting with brokered AWS services using the CLI](https://cloud.gov/docs/services/s3/#interacting-with-your-s3-bucket-from-outside-cloudgov).
+
+The cause of the issue is that AWS CLI is overriding certificates that cloud.gov provides and preventing the AWS CLI from being able to validate TLS connections, thus causing CLI commands to fail.
+
+To force the AWS CLI to use the system CA cert stores, you can add this environment variable:
+
+```shell
+AWS_CA_BUNDLE=/etc/ssl/certs/ca-certificates.crt
+```
+
+Environment variables can be set for your application either by using a [manifest](https://docs.cloudfoundry.org/devguide/deploy-apps/manifest-attributes.html#env-block) or the [`cf set-env` CLI command](https://cli.cloudfoundry.org/en-US/v7/set-env.html).
```

Comment:
Sorry I'm late to provide this feedback: There's an implication here that you should _only_ set the environment variable via manifest or CF CLI. However, since the doesn't change for each app or each environment, it could also be set in the `.profile` for the application itself, or in the Dockerfile used to build an application image. It might make _more_ sense to set the value there since it's effectively immutable.